File: Practica4/QytetetRuby P4/pkg/QytetetRuby-0.0.1/lib/especulador.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
module ModeloQytetet
  class Especulador < Jugador
    
    @@FACTOR_ESPECULADOR = 2
    attr_accessor :FACTOR_ESPECULADOR
    def initialize(jugador , fianza)
      super(jugador.nombre,jugador.encarcelado,jugador.saldo,jugador.casilla_actual,jugador.carta_libertad,jugador.propiedades)
      @fianza = fianza
    end
    
    def pagar_impuestos(cantidad)
      super.modificar_saldo(-cantidad/2)
    end
    
    def ir_a_carcel(casilla)
      no_carcel = pagar_fianza(@fianza)
      if(!no_carcel)
        super.ir_a_carcel(casilla)
      else
        super.modificar_saldo(-@fianza)
      end
     end
    
    def get_factor_especulacion()
      @@FACTOR_ESPECULADOR
    end
    
    
    def convertirme(fianza)
      return self
    end
    
    def pagar_fianza(cantidad)
      if(super.saldo < cantidad)
        resultado = false
      else
        resultado = true
      end
      resultado
    end
    
    def to_s
      "************************************************\n
       #{super.to_s} \n
       Fianza: #{@fianza}"
       puts "hola estoy en especulador"
    end
    
    private :pagar_fianza
    
  end
end

File: Practica4/QytetetJavaP4/src/modeloqytetet/Especulador.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloqytetet;

/**
 *
 * @author Lorena
 */
public class Especulador extends Jugador{
    static final int FactorEspeculador = 2;
    private int fianza;

    protected Especulador(Jugador jugador, int fianza){
        super(jugador);
        this.fianza = fianza;
        //super.setFactorEspeculador(FactorEspeculador);
    }
    
    protected void pagarImpuesto(int cantidad){
        super.modificarSaldo(-cantidad/2);
    }
    
    protected void irACarcel(Casilla casilla){
        boolean noCarcel = pagarFianza(fianza);
        if(!noCarcel){
            super.irACarcel(casilla);
        }
        else{
            super.modificarSaldo(-fianza);
        }
    }
    
    protected Especulador convertirme(int fianza){
        return this;
    }
    
    private boolean pagarFianza(int cantidad){
        if(super.getSaldo() < cantidad){
            return false;
        }
        else{
            return true;
        }
    }
    
    public static int getFactorEspeculador() {
        return FactorEspeculador;
    }
    
    
    
    //Método toString() devuelve un String con el estado del objeto correspondiente
    @Override
    public String toString(){
        return  super.toString() 
                +"Factor especulador: " + FactorEspeculador + "\n"
                + "Fianza: " + fianza + "\n";
    }

   
}
